js-html-games/bogscript.js

```diff
@@ -1,7 +1,378 @@
+            erase();
+            typedInput.pop();
+            console.log(typedInput);
+        }
+        for (let i = 0; i < keyboardLetters.length; i++) {
+            if (event.code === `Key${alphabet[i]}`) {
+                buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor = "rgb(104, 157, 157)";
+                typedInput.push(alphabet[i]);
+                type();
+                console.log(typedInput);
+                console.log(alphabet[i]);
+            }
+        }
+    })
+    document.addEventListener('keyup', function (event) {
+        if (event.code === "Enter") {
+            buttons[buttons.length - 2].style.backgroundColor = "lightblue";
+        }
+        if (event.code === "Backspace") {
+            buttons[buttons.length - 1].style.backgroundColor = "lightblue";
+        }
+        for (let i = 0; i < keyboardLetters.length; i++) {
+            if (event.code === `Key${alphabet[i]}`) {
+                if (buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor !== "grey") {
+                    buttons[keyboardLetters.indexOf(alphabet[i]) + 1].style.backgroundColor = "lightblue";
+                }
+            }
+        }
+    })
+}
+
+function type() {
+    const guessCols = document.querySelectorAll(".letter");
+    if (typedInput.length < 6) {
+        console.log(currentRow);
+        guessCols[(typedInput.length - 1) + (numC * currentRow)].innerText = typedInput[typedInput.length - 1];
+    }
+    else {
+        typedInput = typedInput.slice(0, numC);
+    }
+}
+
+function erase() {
+    const guessCols = document.querySelectorAll(".letter");
+    console.log(typedInput);
+    guessCols[(typedInput.length-1) + (numC * currentRow)].innerText = "";
+}
+
+function reset() {
+    const guessCols = document.querySelectorAll(".letter");
+    currentRow = 0;
+    typedInput = [];
+    for (let i = 0; i < guessCols.length; i++) {
+        error.innerText = "New word!"
+        guessCols[i].innerText = "";
+        setDivColor(i, "rgb(56, 162, 150)")
+    }
+}
+
+function matchWord(ans, userAns, txtList) {
+    const guessCols = document.querySelectorAll(".letter");
+    const buttons = document.querySelectorAll('button');
+    let correctLetters = 0;
+    if (userAns.length === ans.length) {
+        if (txtList.includes(userAns.toLowerCase())) {
+            console.log("next row");
+            for (let i = 0; i < ans.length; i++) {
+                if (userAns[i] === ans[i]) {
+                    setDivColor(`${numC * currentRow + i}`, "rgb(104, 196, 54)");
+                    correctLetters++;
+                }
+                else if (ans.includes(userAns[i])) {
+                    setDivColor(`${numC * currentRow + i}`, "#f0ce7a");
+                }
+                else {
+                    buttons[keyboardLetters.indexOf(userAns[i]) + 1].style.backgroundColor = "grey";
+                    setDivColor(`${numC * currentRow + i}`, "#d14134");
```

Comment:
DRY: set the variable once and re-use so you can "don't repeat yourself"
repeating yourself allows you to have more mistakes as you code, less easy to modify code, and harder to read

also, use plural for arrays, and singular for single element. 

```
const target = String(numC * currentRow + i);
const userAnswer = userAnswers[i];
const answer = answers[i]

if (userAnswer === answer) {
    setDivColor(target, "rgb(104, 196, 54)");
    correctLetters++;
} else if (answer.includes(userAnswer)) {
    setDivColor(target, "#f0ce7a");
} else {
    const buttonIndex = keyboardLetters.indexOf(userAnswer) + 1
    buttons[buttonIndex].style.backgroundColor = "grey";
    setDivColor(target, "#d14134");
}
```